lib/matplotlib/_constrained_layout.py

```diff
@@ -187,7 +187,7 @@ def make_layoutgrids(fig, layoutgrids, rect=(0, 0, 1, 1)):
 
     # for each axes at the local level add its gridspec:
     for ax in fig._localaxes:
-        if hasattr(ax, 'get_subplotspec'):
+        if ax.get_subplotspec():
```

Comment:
These are either `SubplotSpec` or `None`, right?

I think in these cases it's common to explicitly check `... is not None` and not implicitly relying on the Truthiness of `SubplotSpec` instances.

Also the following checks.

Reply:
Yup, also refactored a bit some of the checks as one can sometimes directly call `get_gridspec()` instead of `get_subplotspec().get_gridspec()`.